app/components/add-cron-job.js

```diff
@@ -1,45 +1,68 @@
 import Component from '@ember/component';
 import { inject as service } from '@ember/service';
-import { task } from 'ember-concurrency';
+import { computed } from '@ember/object';
+import { mapBy } from '@ember/object/computed';
 
 export default Component.extend({
   store: service(),
+  branchesFetcher: service(),
 
   classNames: ['form--cron'],
 
+  intervals: ['monthly', 'weekly', 'daily'],
+
+  options: ['Always run', 'Do not run if there has been a build in the last 24h'],
+
+  currentCronJobsBranches: mapBy('repository.cronJobs', 'branch.name'),
+
+  init() {
+    this.reset();
+    this.setProperties({
+      'branchesFetcher.repository': this.get('repository')
+    });
+    this.get('branchesFetcher.showSearchResults').perform();
```

Comment:
I don't think it's a good idea to pre-fetch branches. Currently it feel like there are just 10 branches (because they are pre-fetched). It would be better to show empty list with a placeholder, like "Start typing branch name to search", and then show search results in the list as they load.